src/components/WorkflowEngine/engine.ts

```diff
@@ -0,0 +1,169 @@
+import { performHttpReq, HttpRequestParams } from '~utils/http'
+import { formatError } from '~utils/onfidoApi'
+import type { StepConfig } from '~types/steps'
+import type {
+  WorkflowResponse,
+  OutcomeStepKeys,
+  GetWorkflowFunc,
+  CompleteWorkflowFunc,
+  GetFlowStepFunc,
+  WorkflowStepConfig,
+} from './utils/WorkflowTypes'
+
+export interface EngineInterface {
+  getOutcomeStep(workflow: WorkflowResponse | undefined): OutcomeStepKeys
+  getWorkflow: GetWorkflowFunc
+  completeWorkflow: CompleteWorkflowFunc
+  getWorkFlowStep: GetFlowStepFunc
+}
+
+export type EngineProps = {
+  token: string | undefined
+  workflowRunId: string | undefined
```

Comment:
Better to enforce that the property is a string, the we can remove some guard statements. Why should we have a WorkflowEngine running without an id?